yt/data_objects/selection_objects/data_selection_objects.py

```diff
@@ -262,6 +262,8 @@ def _generate_fields(self, fields_to_generate):
                         fi.units = sunits
                         fi.dimensions = dimensions
                         self.field_data[field] = self.ds.arr(fd, units)
+                    if fi.output_units is None:
```

Comment:
Apparently. I'll admit I don't know what it's meant for

Reply:
Was it meant to be removed at any point ? It's still used in the profiles.py module

Reply:
If I am remembering correctly, the expert on this is @brittonsmith and we should defer to him.  That being said, unless this sticks out at him, I think we're good to go with this fix.

Reply:
My "expertise" on `output_units` is only from the fact that I have been bitten by it the most. I never understood why it existed, especially since all the visualization functions have unit setting. It's only ever caused problems for me. Can we just get rid of it?

Reply:
I opened an issue specifically for this discussion (#3967)